modules/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3Repository.java

```diff
@@ -340,49 +339,14 @@ public void onFailure(Exception e) {
     }
 
     @Override
-    protected SnapshotDeleteListener wrapWithWeakConsistencyProtection(SnapshotDeleteListener listener) {
-        return new SnapshotDeleteListener() {
-            @Override
-            public void onDone() {
-                listener.onDone();
-            }
-
-            @Override
-            public void onRepositoryDataWritten(RepositoryData repositoryData) {
-                logCooldownInfo();
-                final Scheduler.Cancellable existing = finalizationFuture.getAndSet(threadPool.schedule(() -> {
-                    final Scheduler.Cancellable cancellable = finalizationFuture.getAndSet(null);
-                    assert cancellable != null;
-                    listener.onRepositoryDataWritten(repositoryData);
-                }, coolDown, snapshotExecutor));
-                assert existing == null : "Already have an ongoing finalization " + finalizationFuture;
-            }
-
-            @Override
-            public void onFailure(Exception e) {
-                logCooldownInfo();
-                final Scheduler.Cancellable existing = finalizationFuture.getAndSet(threadPool.schedule(() -> {
-                    final Scheduler.Cancellable cancellable = finalizationFuture.getAndSet(null);
-                    assert cancellable != null;
-                    listener.onFailure(e);
-                }, coolDown, snapshotExecutor));
-                assert existing == null : "Already have an ongoing finalization " + finalizationFuture;
-            }
-        };
-    }
-
-    /**
-     * Wraps given listener such that it is executed with a delay of {@link #coolDown} on the snapshot thread-pool after being invoked.
-     * See {@link #COOLDOWN_PERIOD} for details.
-     */
```

Comment:
Do we want to retain this comment and maybe expand it about what the weak consistency protection is?

Reply:
Ah good point I didn't mean to lose that comment.